mantidproject/mantid
# Export SANS Table as CSV

**Description of work.**
Added option to export current table as a csv file. This can then be re-loaded as a batch file afterwards.

**Report to:** nobody

**To test:**
1. Interfaces -> SANS -> SANS v2
2. Load any acceptable user file and batch file
3. Change some of the data in the table
4. Click `Export Table` button, and select filename to save as (should default to loaded batch file)
5. Upon saving, check that process/load/export buttons are disabled and re-enabled afterwards (for a single row table this is possibly too quick to perceive)
6. Load your newly saved batch file and check that the changes you made to the data are present. Some rows are not included while saved out. The rows which are included are sample and can scatter/trans/direct, user file, and output file.

Fixes #23970 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
